src/latest/dbt_yml_files.py

```diff
@@ -0,0 +1,1009 @@
+    post_hook: Optional[ArrayOfStrings] = Field("", alias="post-hook")
+    pre_hook: Optional[ArrayOfStrings] = Field("", alias="pre-hook")
+    quote_columns: Optional[BooleanOrJinjaString] = ""
+    strategy: Optional[str] = ""
+    tags: Optional[StringOrArrayOfStrings] = ""
+    target_database: Optional[str] = ""
+    target_schema: Optional[str] = ""
+    unique_key: Optional[StringOrArrayOfStrings] = ""
+    updated_at: Optional[str] = ""
+
+
+class Quoting(BaseModel):
+    model_config = ConfigDict(
+        extra="forbid",
+    )
+    database: Optional[BooleanOrJinjaString] = ""
+    identifier: Optional[BooleanOrJinjaString] = ""
+    schema_: Optional[BooleanOrJinjaString] = Field("", alias="schema")
+
+
+class Relationships(BaseModel):
+    model_config = ConfigDict(
+        extra="forbid",
+    )
+    name: Optional[str] = ""
+    config: Optional[TestConfigs] = ""
+    field: str = Field(
+        ..., description="The foreign key column", title="Relationships: Field"
+    )
+    to: str = Field(
+        ..., examples=["ref('parent_model')", "source('parent_schema', 'parent_table')"]
+    )
+    where: Optional[str] = ""
+
+
+class RelationshipsTest(BaseModel):
+    relationships: Optional[Relationships] = ""
+
+
+class AcceptedValues(BaseModel):
+    name: Optional[str] = ""
+    config: Optional[TestConfigs] = ""
+    quote: Optional[bool] = ""
+    values: List[str]
+    where: Optional[str] = ""
+
+
+class AcceptedValuesTest(BaseModel):
+    accepted_values: Optional[AcceptedValues] = AcceptedValues(values=[])
+
+
+class NotNull(BaseModel):
+    name: Optional[str] = ""
+    config: Optional[TestConfigs] = ""
+    where: Optional[str] = ""
+
+
+class NotNullTest(BaseModel):
+    not_null: Optional[NotNull] = ""
+
+
+class Unique(BaseModel):
+    name: Optional[str] = ""
+    config: Optional[TestConfigs] = ""
+    where: Optional[str] = ""
+
+
+class UniqueTest(BaseModel):
+    unique: Optional[Unique] = ""
+
+
+class DataTests(
+    RootModel[
+        Union[str, RelationshipsTest, AcceptedValuesTest, NotNullTest, UniqueTest]
+    ]
+):
+    root: Union[str, RelationshipsTest, AcceptedValuesTest, NotNullTest, UniqueTest]
+
+
+class DimensionBase(BaseModel):
+    model_config = ConfigDict(
+        extra="forbid",
+        regex_engine="python-re",
+    )
+    name: str = Field(..., pattern="(?!.*__).*^[a-z][a-z0-9_]*[a-z0-9]$")
+    description: Optional[str] = ""
+    expr: Optional[Union[str, bool]] = ""
+    is_partition: Optional[bool] = True
+
+
+class TimeDimensionType(Enum):
+    TIME = "TIME"
+    time = "time"
+
+
+class CategoricalDimensionType(Enum):
+    CATEGORICAL = "CATEGORICAL"
+    categorical = "categorical"
+
+
+class CategoricalDimension(DimensionBase):
+    type: CategoricalDimensionType
+
+
+class TimeDimension(DimensionBase):
+    type: TimeDimensionType
+    type_params: DimensionTypeParams
+
+
+class Dimension(RootModel[Union[CategoricalDimension, TimeDimension]]):
```

Comment:
Can we write this as a union like this https://github.com/dbt-labs/ai-codegen-api/blob/main/src/codegen/llm_output/semantic_model.py#L66